packages/typegpu-react/src/use-mirrored-uniform.ts

```diff
@@ -0,0 +1,60 @@
+import * as d from 'typegpu/data';
+import { useRoot } from './root-context.tsx';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import type { ValidateUniformSchema } from 'typegpu';
+
+interface MirroredValue<TSchema> {
+  schema: TSchema;
+  readonly $: d.InferGPU<TSchema>;
+}
+
+export function useMirroredUniform<
+  TSchema extends d.AnyWgslData,
+  TValue extends d.Infer<TSchema>,
+>(
+  schema: ValidateUniformSchema<TSchema>,
+  value: TValue,
+): MirroredValue<TSchema> {
+  const root = useRoot();
+  const [uniformBuffer, setUniformBuffer] = useState(() => {
+    return root.createUniform(schema, value);
+  });
+  const prevSchemaRef = useRef(schema);
+  const cleanupRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  useEffect(() => {
+    let currentBuffer = uniformBuffer;
+    if (!d.deepEqual(prevSchemaRef.current as d.AnyData, schema as d.AnyData)) {
+      currentBuffer.buffer.destroy();
+      currentBuffer = root.createUniform(schema, value);
+      setUniformBuffer(currentBuffer);
+      prevSchemaRef.current = schema;
+    }
+
+    currentBuffer.write(value);
```

Comment:
Tiny nit, we don't need to write to buffers that have just been created, as they're initialized with the new value

```suggestion
    if (!d.deepEqual(prevSchemaRef.current as d.AnyData, schema as d.AnyData)) {
      uniformBuffer.buffer.destroy();
      setUniformBuffer(root.createUniform(schema, value));
      prevSchemaRef.current = schema;
    } else {
      uniformBuffer.write(value);
    }
```